src/bert_layers/model.py

```diff
@@ -617,6 +617,11 @@ def __init__(self, config: FlexBertConfig):
         super().__init__(config)
         self.embeddings = get_embedding_layer(config)
         self.encoder = get_encoder_layer(config)
+        if "prenorm" in config.bert_layer:
+            # if we use prenorm attention we need to add a final norm
+            self.final_norm = get_norm_layer(config)
+        else:
+            self.final_norm = None
```

Comment:
I'd rather have this be an option that's defaulted to True rather than hardcoding for prenorm.

Reply:
Ok for me to not hardcode it like this. But given that final norm never should be used with postnorm attention, and always should be used with prenorm attention, I'd be inclined towards not only having a warning, but even an assert for compatibility between config.final_norm and config.bert_layers. Else it is easy to make a mistake when you are changing the config, speaking of my own experience.

Reply:
No hardcoding for future flexibility if something comes up + assert has my vote too.